qiskit/transpiler/synthesis/aqc/aqc.py

```diff
@@ -0,0 +1,108 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""A generic implementation of Approximate Quantum Compiler."""
+from typing import Optional
+
+import numpy as np
+
+from qiskit.algorithms.optimizers import L_BFGS_B, Optimizer
+from qiskit.quantum_info import Operator
+from .approximate import ApproximateCircuit, ApproximatingObjective
+
+
+class AQC:
+    """
+    A generic implementation of Approximate Quantum Compiler. This implementation is agnostic of
+    the underlying implementation of the approximate circuit, objective, and optimizer. Users may
+    pass corresponding implementations of the abstract classes:
+
+        * Optimizer is an instance of :class:`~qiskit.algorithms.optimizer.Optimizer` and
+          used to run the optimization process. A choice of optimizer may affect overall
+          convergence, required time for the optimization process and achieved objective value.
+        * Approximate circuit represents a template which parameters we want to optimize.
+          Currently, there's only one implementation based on 4-rotations CNOT unit blocks:
+          :class:`~qiskit.transpiler.aqc.CNOTUnitCircuit`. See the paper for more details.
+        * Approximate objective is tightly coupled with the approximate circuit implementation
+          and provides two methods for computing objective function and gradient with respect to
+          approximate circuit parameters. This objective is passed to the optimizer. Currently,
+          there's only one implementation based on 4-rotations CNOT unit blocks:
+          :class:`~qiskit.transpiler.aqc.DefaultCNOTUnitObjective`. This is a naive implementation
+          of the objective function and gradient and may suffer from performance issues.
+    """
+
+    def __init__(
+        self,
+        optimizer: Optional[Optimizer] = None,
+        seed: Optional[int] = None,
+    ):
+        """
+        Args:
+            optimizer: an optimizer to be used in the optimization procedure of the search for
+                the best approximate circuit. By default ``L_BFGS_B`` is used with max iterations
+                is set to 1000.
+            seed: a seed value to be user by a random number generator.
+        """
+        super().__init__()
+        self._optimizer = optimizer
+        self._seed = seed or 12345
```

Comment:
it's standard that `None` seed is random, not a fix value.
```suggestion
        self._seed = seed
```